fuel-vm/src/state/debugger.rs

```diff
@@ -14,31 +14,37 @@ use std::collections::{
     HashSet,
 };
 
-#[derive(Debug, Default, Clone)]
 /// Debugger implementation for the VM.
-///
-/// Required features:
-/// - `debug`
+#[derive(Debug, Default, Clone)]
 pub struct Debugger {
+    /// Debugger is active and used.
+    is_active: bool,
     /// Single-stepping mode triggers a breakpoint after each instruction
     single_stepping: bool,
     breakpoints: HashMap<ContractId, HashSet<Word>>,
     last_state: Option<ProgramState>,
 }
 
 impl Debugger {
+    /// Returns `true` if the `Debugger` is active and used.
+    pub const fn is_active(&self) -> bool {
```

Comment:
Would it be useful to expose an API to activate the debugger explicitly, without calling a `set_*` function?

Reply:
Is there any way to turn off debugging? In case you need to switch between debugging and performance modes